testtools/matchers/_warnings.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2009-2016 testtools developers. See LICENSE for details.
+
+__all__ = [
+    'Warnings',
+    'warning_message',
+    'is_deprecated']
+
+import warnings
+
+from ._basic import Is
+from ._const import Always
+from ._datastructures import MatchesListwise, MatchesStructure
+from ._higherorder import (
+    AfterPreprocessing,
+    Annotate,
+    MatchesAll,
+    Not,
+    )
+from ._impl import Mismatch
+
+
+def warning_message(category_type, message=None, filename=None, lineno=None,
+                    line=None):
+    """
+    Create a matcher that will match `warnings.WarningMessage`\s.
+
+    :param category_type: A warning type.
+    :param message_matcher: Match the warning message against this.
+    :param filename_matcher: Match the warning filename against this.
+    :param lineno_matcher: Match the warning line number against this.
+    :param line_matcher: Match the warning line of code against this.
```

Comment:
I think it may be confusing that some of these parameters are types, and some are matchers. There are a few things we could do in the docstring to make this more clear:

 * An example would help people understand how this function is intended to be used.
 * The individual :param: lines could be clearer. For example:

 > :param message_matcher: A matcher object that will be evaluated against the message category.